app/components/Page/index.js

```diff
@@ -1,23 +1,35 @@
 import React from 'react'
+import classNames from 'classnames/bind'
 
 import Header from 'components/Header'
-import WhitelistedDappState from 'components/WhitelistedDappState'
 import styles from './index.css'
 
-const Page = (props) => (
-  <div>
-    {!props.withoutHeader &&
-      <Header
-        account={props.account}
-        logOut={props.logOut}
-        settings={props.settings}
-      />
-    }
-    {props.whitelist && <WhitelistedDappState />}
-    <div className={styles.container}>
-      {props.children}
+const cx = classNames.bind(styles)
+
+const Page = ({
+  children,
+  padding,
+  ...props,
```

Comment:
That is dangerous. You are passing here props without a name. As far as I see they are used on the Header component.

I suggest transforming the Header component into something like [this](https://github.com/gnosis/gnosis-team-safe/tree/master/src/components/Header)

In that way to do not overload the Page component with props only related to Header component, and it is the Header the responsible of fetching stuff from the redux store using its own selectors.

Also, your project is getting bigger, I strongly recommend to introduce a type checker (Flow is my choice).